R/utilities.R

```diff
@@ -223,3 +223,62 @@ is_windows <- function() {
     sysname <- Sys.info()["sysname"]
     return(sysname == "Windows")
 }
+
+#' `expand_time_grid`
+#'
+#' This function expands a given time grid by setting a default grid if one hasn't been provided
+#' and then verifying it's properties.
+#' The grid must be finite, sorted, and contain unique values.
+#'
+#' @param time_grid (`numeric` or `NULL`)\cr A vector of times which quantities will be
+#' evaluated at. If NULL, a default grid will be created as a length 201 vector spanning
+#' from 0 to `max_time`.
+#' @param max_time (`numeric``)\cr Specifies the maximum time to be used in creating the default grid.
+#' @return Returns the expanded time_grid.
+#' @keywords internal
+expand_time_grid <- function(time_grid, max_time) {
+    default_grid <- seq(from = 0, to = max_time, length = 201)
+    if (is.null(time_grid)) {
+        time_grid <- default_grid
+    }
+    assert_that(
+        !any(is.na(time_grid)),
+        is.numeric(time_grid),
+        !is.null(time_grid),
+        !is.unsorted(time_grid),
+        !any(duplicated(time_grid)),
+        all(is.finite(time_grid)),
+        msg = "`time_grid` needs to be finite, sorted, unique valued numeric vector"
+    )
+    return(time_grid)
```

Comment:
minor style comment: best to avoid explicit `return()` statements if not needed, e.g. here just
```suggestion
    time_grid
```